src/main/kotlin/StringAdditionCalculator.kt

```diff
@@ -0,0 +1,18 @@
+class StringAdditionCalculator {
+    fun sum(input: String): Int {
+        val (delimiters, expression) = parseDelimitersAndExpression(input)
+        val tokens = expression.split(delimiters.toRegex())
+        val numbers = tokens.map { it.toIntOrNull() ?: throw IllegalArgumentException("Invalid input: $it") }
+        return numbers.sum()
+    }
+
+    private fun parseDelimitersAndExpression(input: String): Pair<String, String> {
+        return if (input.startsWith("//")) {
```

Comment:
Please extract this magic number as a constant for better readability and maintainability.